### DISCUSSION TITLE
Parse error on indirect object reference

### DISCUSSION THREAD

**Discussion by stollcri** [MatthiasValvekens/pyHanko#494] [Q&A]
I have some PDF files - ones already signed by Jotform - that cause the following error to be raised. There seems to be an invalid reference in the PDF files; when I catch the error on line 279 of the same file and handle it (ignore that missing reference by naively just returning `BooleanObject(False)`) I can load and manipulate the file as expected.

I know that is not the correct fix. My question is, how can I debug this error to get more specific information about the conditions causing the problem, so that I can implement the correct fix?

I used an old tool called 'peepdf' and it suggested that the stream location referenced by the raised error was inside of an embedded image, which doesn't' seem correct.

```
raise PdfReadError(
    f"Parse error on indirect object reference around {pos}"
)
```
https://github.com/MatthiasValvekens/pyHanko/blob/6da540450eee75a00d7379a3b202bd31c2fb9922/pyhanko/pdf_utils/generic.py#L616C1-L618C14

**Comment by MatthiasValvekens** [comment]
Unless you can share the file in question, I'm afraid that there's not much to say about this :).

PDF syntax is already quite fungible, and while pyHanko's parser is indeed a bit pickier than most, allowing malformed object refs (and e.g. defaulting them to `latest`(doesn't really sit well with me for various reasons... Mainly the risk of accidentally introducing parsing ambiguities.

**Reply by stollcri** [reply]
Here is an example document.

Yeah, I don't want to suggest any changes that would cause fragility. I just wanted to dig in, validate the problem, and see if there are any potential enhancements; I just wasn't sure of the best way to debug the issue. 

[2024-11-15.pdf](https://github.com/user-attachments/files/17817332/2024-11-15Evaluation.PTaa7fe7a7-a489-41d0-9334-3ad2ba712358.pdf)
